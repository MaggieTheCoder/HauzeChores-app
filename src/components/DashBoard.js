/* eslint-disable no-unused-vars */
import React, { useEffect, useState } from "react";
import propTypes from "prop-types";
// import AddHouse from "./AddHouse";
import AddTask from "./AddTask";

import Tabs from "./Tabs";
import "../styles/Dashboard.css";
import getAllHouseTasks from "../requests/getAllHouseTasks";
import getAllUserTasks from "../requests/getAllUserTasks";

const DashBoard = ({ houseId, userId, setUserId }) => {
  const [houseTasks, setHouseTasks] = useState([]);
  const [userTasks, setUserTasks] = useState([]);
  const [addedANewTask, setAddedANewTask] = useState(1);

  useEffect(() => {
    console.log("page renderig", addedANewTask);
    getAllHouseTasks(houseId, setHouseTasks, userId, setUserTasks);
    getAllUserTasks(userId, setUserTasks);
  }, [addedANewTask]);

  return (
    <div className="dashboard">
      <p>user: {userId}</p>
      <p>house: {houseId}</p>
      <p>new task: {addedANewTask}</p>
      {/* <AddHouse /> */}
      <AddTask
        houseId={houseId}
        userId={userId}
        setAddedANewTask={setAddedANewTask}
      />
      <Tabs
        houseTasks={houseTasks}
        userTasks={userTasks}
        setAddedANewTask={setAddedANewTask}
        userId={userId}
      />
    </div>
  );
};

export default DashBoard;

DashBoard.propTypes = {
  houseId: propTypes.number.isRequired,
  userId: propTypes.number.isRequired,
  setUserId: propTypes.func.isRequired,
};
