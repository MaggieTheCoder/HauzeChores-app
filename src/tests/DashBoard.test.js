import React from "react";
import { render } from "@testing-library/react";
import DashBoard from "../components/DashBoard";

describe("Dashboard component", () => {
  const validProps = {
    houseId: 1,
    userId: 1,
    setUserId: () => {},
  };
  test("Component renders as expected", () => {
    const { asFragment } = render(
      <DashBoard
        houseId={validProps.houseId}
        userId={validProps.userId}
        setUserId={validProps.setUserId}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
