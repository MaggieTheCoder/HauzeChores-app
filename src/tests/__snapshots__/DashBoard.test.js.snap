// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component Component renders as expected 1`] = `
<DocumentFragment>
  <div
    class="dashboard"
  >
    <p>
      user: 1
    </p>
    <p>
      house: 1
    </p>
    <p>
      new task: 1
    </p>
    <div
      class="add-task"
    >
      <form
        class="task__list"
      >
        <label
          for="type"
        >
          Task
          <select
            id="type"
            name="type"
          >
            <option
              value="Empty Dishwasher"
            >
              Empty Dishwasher
            </option>
            <option
              value="Hoover living room"
            >
              Hoover living room
            </option>
            <option
              value="Water the plants"
            >
              Water the plants
            </option>
            <option
              value="custom"
            >
              Add your own
            </option>
          </select>
        </label>
        <button
          type="button"
        >
          Add
        </button>
      </form>
    </div>
    <div
      class="add-team"
    >
      <form>
        Invite
        <input
          class="input__box"
          placeholder="email"
          type="text"
        />
        <button
          type="button"
        >
          Send
        </button>
      </form>
    </div>
    <div
      class="tabs"
    >
      <button
        class="tab active"
        type="button"
      >
        All Tasks 
      </button>
      <button
        class="tab "
        type="button"
      >
        My Tasks 
      </button>
      <button
        class="tab "
        type="button"
      >
        Leaderboard 
      </button>
    </div>
    <div
      class="panels"
    >
      <div
        class="panel active"
      />
    </div>
  </div>
</DocumentFragment>
`;
